### ISSUE TITLE
Error loading wms layer

### ISSUE THREAD

**Issue by kiri03** [opengisch/QField#5497]
### Describe the issue
I have a project in QGIS where everything is displayed correctly, including several WMS layers, but when I send that project to QField there is a specific layer that is not displayed and an error related to the SSL certificate appears:

Map request failed [error: SSL handshake failed: The issuer certificate of a locally looked up certificate could not be found url: https://ovc.catastro.meh.es/Cartografia/WMS/ServidorWMS.aspx?

I have deleted all the layers leaving only the layer with the WMS layer and still nothing is displayed in QField, the same error appears.
I attach the QGZ file with the WMS layer to be able to reproduce the error in QField and see if someone can provide me with a solution.)

[Catastro_Spain.zip](https://github.com/user-attachments/files/16451772/Catastro_Spain.zip)

### Desktop (please complete the following information)
 - OS: Windows 11 23H2
 - QGIS Version 3.38.1
 - QFieldSync Version 4.9.1

### Mobile (please complete the following information)
 - Device: Realme GT
 - OS: Android 13
 - QField version: 3.3.7 - Darién

**Comment by nirvn** [comment]
@kiri03 , when I load this locally in QGIS, I get the following dialog:

![image](https://github.com/user-attachments/assets/104de296-2c49-4c8f-8c1a-d429895f22b6)

These custom certificates aren't supported by QField (yet).

**Comment by nirvn** [comment]
Loading the URL on a browser also reveals a problematic certificate:

![image](https://github.com/user-attachments/assets/2c554f2d-7fa1-4a6d-97ae-1dc6a776ada6)

**Comment by kiri03** [comment]
I don't understand it.
When I open that project in QGIS it opens perfectly without any error message.
![Captura](https://github.com/user-attachments/assets/02e87b78-6c70-4dda-a885-dee1de5e995b)

**Comment by nirvn** [comment]
@kiri03 , have you tried a/ opening the URL in your browser, and b/ opening the project in a _new QGIS profile_? It's possible you accepted the custom certificate a long time ago.

**Comment by kiri03** [comment]
I tried putting that link on a new computer and no error appears.
This is what appears:
![Captura2](https://github.com/user-attachments/assets/44affc84-fa05-4a11-984b-85cae076d4c5)

**Comment by nirvn** [comment]
@kiri03 , are you typing in the http*s* URL?

**Comment by kiri03** [comment]
This link:
https://ovc.catastro.meh.es/Cartografia/WMS/ServidorWMS.aspx

**Comment by nirvn** [comment]
@kiri03 , OK, I see it works with Chrome but not with Firefox. Curious.

**Comment by kiri03** [comment]
No, I'm not using VPN.
Te project open perfectly in QGIS but not in QField.

**Comment by nirvn** [comment]
@kiri03 , ok. What version of QGIS are you using?

**Comment by kiri03** [comment]
QGIS 3.38.1
all the information is indicated at the beginning of this thread

**Comment by kiri03** [comment]
QField 3.3.8

**Comment by kiri03** [comment]
I hope this issue will be fixed in a future version of QField.

**Comment by kiri03** [comment]
QField 3.3.9 released and the problem persist with this layers WMS

**Comment by m-kuhn** [comment]
Looks like mozilla has accepted Firmaprofesional 2 months ago https://bugzilla.mozilla.org/show_bug.cgi?id=1785215

**Comment by nirvn** [comment]
@m-kuhn , good catch. We'll have to wait until curl's CA certificates extract to catch up then (see https://curl.se/docs/caextract.html).

I will say though that I'm on Firefox nightly, and that WMS server is still giving me a SEC_ERROR_UNKNOWN_ISSUER error code.

**Comment by m-kuhn** [comment]
Interesting as it says

> Successfully tested in Nightly 128.0a1 (2024-06-09) (64-bit)

<img width="757" alt="Screenshot 2024-08-10 at 11 53 41" src="https://github.com/user-attachments/assets/7cf30df3-d9e3-4536-9353-c49937d147ad">

**Comment by m-kuhn** [comment]
@kiri03 ideally you can reach out to the WMS provider and inform them that their SSL certificate is not accepted by mozilla / on firefox.

<img width="1084" alt="Screenshot 2024-08-10 at 12 00 06" src="https://github.com/user-attachments/assets/972f216a-0783-40f0-a610-c6e90022976a">

The only thing which could potentially be done on QField side is to add an SSL exception management.

**Comment by m-kuhn** [comment]
For reference, this is the test page for firmaprofesional and it is working: https://testsslev2022ec.firmaprofesional.com/ so it has been added, but did not fix the issue here.

The cert has been signed by a different CA (visible in the security details of both pages). This is definitely best solved through `catastro.meh.es` directly.

**Comment by kiri03** [comment]
I don't quite understand the proposed solution.
Does that mean that the problem cannot be solved from within Qfield's own development by modifying the program?

**Comment by m-kuhn** [comment]
This means that this problem is best solved on server side and therefore best if you could make the Spanish authorities aware of that. The fact that it's not working in Firefox -- which is a major, state of the art web browser -- should be good enough for their technical staff to assess the source of the problem.

The problem could also be circumvented from within QField but it requires a non trivial amount of work.

**Comment by kiri03** [comment]
Thank you very much for the reply. Now everything is clear.

**Comment by kiri03** [comment]
I just installed Mozilla Firefox and tried the address:
https://ovc.catastro.meh.es/Cartografia/WMS/ServidorWMS.aspx?

I don't get any SSL certificate error and the page opens normally, just like in Chrome or Edge:

![Imagen3](https://github.com/user-attachments/assets/39f07730-30c4-4c53-aa46-af85a0704151)

**Comment by m-kuhn** [comment]
It also works for me on linux, but not on macos with firefox nightly.
I guess on linux it uses a different set of CA.

**Comment by nirvn** [comment]
@m-kuhn , to add to the mystery: it does _not_ work for me on linux (using Firefox nightly) on Ubuntu 24.04.

**Comment by kiri03** [comment]
> This means that this problem is best solved on server side and therefore best if you could make the Spanish authorities aware of that. The fact that it's not working in Firefox -- which is a major, state of the art web browser -- should be good enough for their technical staff to assess the source of the problem.
> 
> The problem could also be circumvented from within QField but it requires a non trivial amount of work.


I contacted the cadastre registry service in Spain (https://www.sedecatastro.gob.es/) and the answer I got was that TLS 1.0 and 1.1 have recently been disabled.
I don't know exactly what this means and if there is a simple solution.
Thanks

**Comment by m-kuhn** [comment]
Interesting, I just checked with Firefox 130.0.1

- https://ovc.catastro.meh.es/Cartografia/WMS/ServidorWMS.aspx? : Does show a warning *Firefox detected a potential security threat and did not continue to ovc.catastro.meh.es. If you visit this site, attackers could try to steal information like your passwords, emails, or credit card details.*
- https://tls.support/ : shows *Your browser is using TLS 1.3. This is the latest version of TLS, which has no known issues.*

**Comment by nirvn** [comment]
@kiri03 , has the situation improved for you here?

**Comment by kiri03** [comment]
> [@kiri03](https://github.com/kiri03) , has the situation improved for you here?

It seems that with the latest version of QField, the problem is solved, and the WMS cadastral layers load and display perfectly on mobile devices.
Thanks
